spec/models/version_spec.rb

```diff
@@ -571,4 +571,31 @@ def save_all_projects
       end
     end
   end
+
+  describe "order by name" do
+    let(:project) { create(:project) }
+    let(:names) do
+      [
+        "1. xxxx",
+        "1.1. aaa",
+        "1.1. zzz",
+        "1.2. mmm",
+        "1.10. aaa",
+        "9",
+        "10.2",
+        "10.10.2",
+        "10.10.10",
+        "aaaaa",
+        "aaaaa 1."
+      ]
+    end
+    let!(:versions) { names.map { |name| create(:version, name:, project:) } }
+
+    subject { described_class.order(:name, id: :desc).to_a }
+
+    it "returns the versions in semver order" do
+      expect(subject)
+        .to eql versions
+    end
```

Comment:
This spec seems to be deceptive, the names are already ordered in the right order, and also matching the order of ids for the versions created. This can be a source of a false positive expectation. I would order the name randomly when creating the versions, additionally add a spec with the reverse order too.

Also, the `id: :desc` part does not seem to be tested, is it important to have it in `subject { described_class.order(:name, id: :desc).to_a }` ?

```suggestion
    let!(:versions) { names.shuffle.map { |name| create(:version, name:, project:) } }

    subject { described_class.order(:name, id: :desc).to_a }

    it "returns the versions in semver order" do
      expect(subject)
        .to eql versions
    end
    
    describe "descending" do
      subject { described_class.order(name: :desc).to_a }

      it "returns the versions in semver order" do
      expect(subject)
        .to eql versions.reverse
    end
```

Reply:
I see your concern, your suggestion will not work exactly, but I'll try something. The `id: :desc` part is to ensure that if name for ordering is considered identical, then id will be used, so it ensures that name is not ignored, but wouldn't be needed if versions are created shuffled